app/controllers/admins/panel_controller.rb

```diff
@@ -10,6 +10,22 @@ def index
     @drivers = Driver.order('active DESC').all
   end
 
+  def edit_driver_photo
+    @driver = Driver.find_for_authentication(id: params[:id])
+    if request.post?
+      if params[:submit] == 'Save'
+        @driver.avatar = params[:avatar] unless params[:avatar].nil?
+      elsif params[:submit] == 'Delete'
+        @driver.avatar = nil
+      end
+      @driver.save
+      html_body = '<script>window.opener.update_driver_avatar('
```

Comment:
It's a very bad practice to render html with js directly in controller :confounded: